src/main/java/org/jenkinsci/plugins/scriptler/builder/ScriptlerBuilder.java

```diff
@@ -91,19 +96,19 @@ public ScriptlerBuilder(
             @CheckForNull String builderId,
             @CheckForNull String scriptId,
             boolean propagateParams,
-            Parameter[] parameters) {
-        this(builderId, scriptId, propagateParams, Arrays.asList(Objects.requireNonNull(parameters)));
+            @CheckForNull Parameter[] parameters) {
+        this(builderId, scriptId, propagateParams, parameters == null ? List.of() : List.of(parameters));
     }
 
     @DataBoundConstructor
     public ScriptlerBuilder(
             @CheckForNull String builderId,
             @CheckForNull String scriptId,
             boolean propagateParams,
-            @NonNull List<Parameter> parameters) {
+            @CheckForNull List<Parameter> parameters) {
         this.builderId = builderId;
         this.scriptId = scriptId;
-        this.parameters = new ArrayList<>(parameters);
+        this.parameters = parameters == null ? List.of() : List.copyOf(parameters);
```

Comment:
Do not do this. Rather create a new `ArrayList` for the internal field used in the serial form.